### PR TITLE
ppx_js: behave better with merlin

### PR DESCRIPTION
**Pull request by trefis**

### CODE REVIEW

**Comment by TyOverby** [comment]
@trefis please run `make fmt` 

The failure in CI is due to ocamlformat not being applied correctly to your changes

**Comment by hhugo** [comment]
I'll try to run some tests later this weekend.

**Comment by trefis** [comment]
Note: don't merge yet, I think this might need a bit more work :(

**Comment by trefis** [comment]
Nevermind: I had some test failure, but I wasn't trying this branch... :see_no_evil:
This should be fine to merge.

**Comment by hhugo** [comment]
`merlin.hide` seems to be hiding/ignoring too much information.
In particular, asking the type of the right hand side of `##.` or `##` doesn't seems to work.
By rhs, I mean `xxx` in expressions bellow 
`o##xxx`
`o##xxx a b c`
`o##.xxx`
`o##.xxx := e`

**Comment by trefis** [comment]
I have just pushed a new version which should behave as well as the current ppx (I have even noticed some improved answers).

**Comment by trefis** [comment]
@hhugo: CI is green.

File: ppx/ppx_js/tests/ppx.mlt [hunk 1]
```diff
@@ -104,7 +104,7 @@ fun (obj : < > Js.t) -> obj##m
 
 [%%expect
 {|
-Line _, characters 24-27:
```
Comment (hhugo) [inline ppx/ppx_js/tests/ppx.mlt]: This seems to be a regression. `obj`  has type `<  >` not `obj##m`

File: ppx/ppx_js/tests/ppx.mlt [hunk 2]
```diff
@@ -169,7 +169,7 @@ fun (obj : < m : 'a. 'a -> unit Js.meth > Js.t) ->
 
 [%%expect
 {|
-Line _, characters 2-5:
```
Comment (hhugo) [inline ppx/ppx_js/tests/ppx.mlt]: Same issue here

File: ppx/ppx_js/tests/ppx.mlt (hunk 1 above)
Comment (trefis) [reply ppx/ppx_js/tests/ppx.mlt]: Fixed.

File: ppx/ppx_js/tests/ppx.mlt (hunk 2 above)
Comment (trefis) [reply ppx/ppx_js/tests/ppx.mlt]: Fixed.

**Comment by hhugo** [comment]
Thanks

**Pull request merged**